src/ocean/text/xml/Document.d

```diff
@@ -610,9 +607,14 @@ version (Filter)
                 public void*            user;           /// open for usage
                 package Document        doc;            // owning document
                 package XmlNodeType     id;             // node type
-                package MutT[]          prefixed;       // namespace
-                package MutT[]          localName;      // name
-                package MutT[]          rawValue;       // data value
+
+                private MutT[]          prefixedBuff;   // namespace buffer
+                private MutT[]          localNameBuff;  // name buffer
+                private MutT[]          rawValueBuff;   // data value buffer
+
+                package T[]             prefixed;       // namespace
```

Comment:
I looked into this a bit more carefully on a local checkout and have one immediate question - is there any reason why these 3 values have to be slices? Alternative would be to actually copy data into them, i.e. ` copy(rawValue, q [0 .. p - q])`. It feels a bit weird to have both slice and a buffer.

Reply:
Yes, there is. The reason is that the slices are usually a slice into the original, const string and are extremely lightweight.
But, if the values are modified, there needs to be a buffer where the new values can be stored.
If we use the same member for both slice and buffer, the buffers won't be reused.

Reply:
A nicer design might be to create a `ConcatBuffer` in the `Document`, and use that single buffer to store all of the modified pieces. But that's a rather more intrusive change in this PR, where I'm really just trying to fix a mistake in the D2 conversion of `PullParser`. I've never used `Document` so I'd prefer to minimize changes to it.

Reply:
I am suspicious that performance penalty for copying each of those fields into buffer is negligible on the scale of the whole XML parser. Would be interesting to benchmark.

But you are right of course that right now priority should be fixing original porting with minimal involved changes.